tuxemon/technique.py

```diff
@@ -297,25 +302,34 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
-        for stat, slug in zip(statsmaster, statslugs):
+        newstatvalue = 0
+        for stat, slugdata in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
-            override = stat['overridetofull']
-            basestatvalue = getattr(target, slug)
-            if value > 0 and override == False:
-                if dividing == False:
-                    newstatvalue = basestatvalue + value
-                else:
-                    newstatvalue = basestatvalue // value
-                success = True
-                setattr(target, slug, newstatvalue)
-            if override == True and slug == 'hp':
-                success = True
-                target.current_hp = target.hp
+            value = stat.get('value')
+            max_deviation = stat.get('max_deviation')
+            operation = stat.get('operation')
```

Comment:
`operation` and `overridetofull` should have default values different from `None`.

Reply:
It is easier than that: you just have to pass the default value as the second parameter of `get`.